Governance.md

```diff
@@ -0,0 +1,78 @@
+# Governance of the rules_scala
+
+These bazel rules for scala are under the bazelbuild org but governed by a
+contributors from the community.
+
+## Goals of this document
+
+Clarify:
+0. values of the project.
+1. how to get a PR merged
+2. how long you should need to wait for a review
+3. how to resolve conflicts in project direction
+4. how to make larger and disruptive changes
+
+## Values of rules_scala
+
+1. Correctness: bazel values fully reproducible builds which are reliable and hermetic (only depend on the current state of your repository)
+2. Speed: we want to build as fast as possible. In a large repository, we believe caching and remote parallel builds are the only way to achieve this. Without correctness, we can't cache and use remote builds, so correctness enables speed. We will almost never trade off correctness for speed.
+3. Usability: we want it to be as easy as possible to adopt and to maintain a build with these rules. Usability is a lower priority than correctness or speed. We will virtually never trade either correctness or speed for more usability. Usability includes stability: we seek to absolutely minimize breaking changes. We will prefer to keep some warts or deprecated methods than force users to constantly make changes to keep their builds on the latest version of the rules.
```

Comment:
>Usability is a lower priority than correctness or speed.
I'm not sure usability being the lowest priority item is pragmatic or a value this project should adhere to. If you build the world's best widget, but very few people can use it, have you succeeded?

>Usability includes stability: we seek to absolutely minimize breaking changes.
I would rather us establish a process for breaking changes: versioned releases, migration guides, etc. Breaking changes will occasionally be necessary and while you can keep them at an absolute minimum by keeping technical debt around, it's not always a good choice.

Reply:
James can you make a proposal? Would you prefer usability be ahead of correctness or speed? I think we should be explicit about our priorities.

This is not an effort to compete with sbt, frankly... we are very far from the ease of use there. I am pretty -1 not following bazel's lead on that {fast, correct}-choose two. We didn't invent that idea, that is the bazel model. We are just affirming in this doc, that priority. Obviously we want it to be usable, but `bazel clean` (which you have to run when you don't have correctness) is very bad for usability. It hammers performance in large repos when users do that with any regularity, which you train them to do when you don't have correctness.

Reply:
I'm a fan of Andy's proposal to this section in https://github.com/bazelbuild/rules_scala/pull/825#discussion_r315457732

Reply:
I agree with Oscar that we need to be clear about priorities of values. Values conflict and when they conflict a project needs to have a compass. Making it vague won't help us IMHO